compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json

```diff
@@ -4180,6 +4180,60 @@
                 }
             ]
         },
+        {
+            "name": "GroupByClauseNode",
+            "base": "IntermediateClauseNode",
+            "kind": "GROUP_BY_CLAUSE",
+            "attributes": [
+                {
+                    "name": "groupKeyword",
+                    "type": "Token"
+                },
+                {
+                    "name": "byKeyword",
+                    "type": "Token"
+                },
+                {
+                    "name": "groupingKey",
+                    "type": "Node",
+                    "occurrences": "MULTIPLE_SEPARATED"
+                }
+            ]
+        },
+        {
+            "name": "GroupingKeyVarNameNode",
+            "base": "Node",
+            "kind": "GROUPING_KEY_VAR_NAME",
+            "attributes": [
+                {
+                    "name": "variableName",
+                    "type": "IdentifierToken"
+                }
+            ]
+        },
```

Comment:
We don't need separate node to hold the variableName right @lochana-chathura WDYT?